### PR TITLE
website_product_delivery_method

### PR DESCRIPTION
**Pull request by TimLai125**

### CODE REVIEW

**Comment by TimLai125** [comment]
@yostashiro If there is no any special concerns, I will merge the PR

File: website_product_delivery_method/data/customer_group_data.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data noupdate="1">
```
Comment (yostashiro) [inline website_product_delivery_method/data/customer_group_data.xml]: Remove `data` tag and make `odoo` tag `<odoo noupdate="1">`.

File: website_product_delivery_method/models/customer_group.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+from odoo import fields, models
+
+
+class CustomerGroup(models.Model):
+    _name = 'customer.group'
+
+    name = fields.Char(
+        string="Name",
+        required=True,
+        copy=False,
```
Comment (yostashiro) [inline website_product_delivery_method/models/customer_group.py]: Very minor point, but why `copy=False`?  If there is only one field in this model and someone wants to copy it, then doesn't that mean the value should be carried to the new record (with "copy" attached at the end)?

File: website_product_delivery_method/models/product_category.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ProductCategory(models.Model):
+    _inherit = 'product.category'
+
+    free_delivery = fields.Boolean(
+        string="Free Delivery Fee",
```
Comment (yostashiro) [inline website_product_delivery_method/models/product_category.py]: Isn't just "Free Delivery" better?

File: website_product_delivery_method/models/sale_order.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        if available_carriers:
+            partner_id = self.partner_id
+
+            # if customer group set on partner then filter delivery method
+            # with matching of partner customer group
+            if partner_id.customer_group:
+                available_carriers = available_carriers.filtered(
+                    lambda d: d.customer_group == partner_id.customer_group)
+
+            # if all order lines are belongs from free delivery product
```
Comment (yostashiro) [inline website_product_delivery_method/models/sale_order.py]: "if all order lines have a free delivery product ..."

File: website_product_delivery_method/models/sale_order.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            # category then return only free delivery methods
+            non_delivery_line = self.order_line.filtered(lambda i : not
+            i.is_delivery)
+            if all(i.product_id.categ_id.free_delivery for i in
+                   non_delivery_line):
+                available_carriers = available_carriers.filtered(
+                    lambda i:i.fixed_price == 0.0)
```
Comment (yostashiro) [inline website_product_delivery_method/models/sale_order.py]: `lambda i: i.delivery_type == 'fixed' and i.fixed_price == 0.0)` to be safe.

File: website_product_delivery_method/models/sale_order.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+            else:
+                available_carriers = available_carriers.filtered(
+                    lambda i: i.fixed_price != 0.0)
```
Comment (yostashiro) [inline website_product_delivery_method/models/sale_order.py]: `lambda i: i.delivery_type != 'fixed' or i.fixed_price != 0.0)`

File: website_product_delivery_method/views/customer_group_view.xml [hunk 7]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0"?>
+<odoo>
+    <data>
```
Comment (yostashiro) [inline website_product_delivery_method/views/customer_group_view.xml]: Remove data tag.

File: website_product_delivery_method/views/customer_group_view.xml [hunk 8]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0"?>
```
Comment (yostashiro) [inline website_product_delivery_method/views/customer_group_view.xml]: Can you check if `encoding="utf-8"?` doesn't have to be explicitly included in 10.0?

File: website_product_delivery_method/views/customer_group_view.xml [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+            <field name="name">customer.group.form</field>
+            <field name="model">customer.group</field>
+            <field name="arch" type="xml">
+                <form string="Customer Group">
+                    <sheet string="Customer Group">
+                         <group>
+                             <group>
+                                <field name="name"/>
+                             </group>
+                             <group/>
```
Comment (yostashiro) [inline website_product_delivery_method/views/customer_group_view.xml]: Remove this.

File: website_product_delivery_method/views/delivery_carrier_view.xml [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+                <field name="customer_group" widget="selection"/>
+            </xpath>
+        </field>
+    </record>
+
+    <record id="view_delivery_carrier_search_inherit_customer_group"
+            model="ir.ui.view">
+        <field name="model">delivery.carrier</field>
+        <field name="name">delivery.carrier.search.inherit.customer
+            .group</field>
```
Comment (yostashiro) [inline website_product_delivery_method/views/delivery_carrier_view.xml]: Do not change lines.

File: website_product_delivery_method/__manifest__.py [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+    'data':[
+        'data/customer_group_data.xml',
+        'security/ir.model.access.csv',
+        'views/customer_group_view.xml',
+        'views/res_partner_view.xml',
+        'views/delivery_carrier_view.xml',
+        'views/product_category_view.xml',
+    ],
+    'installable': True,
+    'application': False,
```
Comment (yostashiro) [inline website_product_delivery_method/__manifest__.py]: Let's remove this line.

**Review by yostashiro — CHANGES_REQUESTED** [review]

File: website_product_delivery_method/views/customer_group_view.xml (hunk 8 above)
Comment (TimLai125) [reply website_product_delivery_method/views/customer_group_view.xml]: I think it is actually related to XML version, it should be UTF-8 or UTF-16 by default if the encoding is not explicitly defined

**Pull request merged**